jni/test/jarray_test.dart

```diff
@@ -186,18 +186,18 @@ void main() {
     array.delete();
   });
   test("Java string array", () {
-    final array = JniArray(JniString.type, 3);
+    final array = JArray(JString.type, 3);
     expect(array.length, 3);
-    array[0] = "حس".jniString();
-    array[1] = "\$".jniString();
-    array[2] = "33".jniString();
+    array[0] = "حس".toJString();
```

Comment:
Since all of these are references, they rely on the attached finalizer and GC to be cleaned up.

It would be cleaner if we would eagerly clean up themselves. Consider using an Arena. (Also the other tests.)

I presume users will start looking at our examples and tests for how to use it, so lets teach them to clean up after themselves.

```dart
  test("Java string array", () {
    using((arena) {
      final array = JArray(JString.type, 3)..deletedIn(arena);
      expect(array.length, 3);
      array[0] = "حس".toJString()..deletedIn(arena);
      array[1] = "\$".toJString()..deletedIn(arena);
      array[2] = "33".toJString()..deletedIn(arena);
      expect(array[0].toDartString(), "حس");
      expect(array[1].toDartString(), "\$");
      expect(array[2].toDartString(), "33");
      array.setRange(
        0,
        3,
        [
          "44".toJString(),
          "55".toJString()..deletedIn(arena),
          "66".toJString()..deletedIn(arena),
          "77".toJString()..deletedIn(arena),
        ],
        1,
      );
      expect(array[0].toDartString(), "55");
      expect(array[1].toDartString(), "66");
      expect(array[2].toDartString(), "77");
      expect(() {
        final _ = array[-1];
      }, throwsRangeError);
      expect(() {
        array[-1] = "44".toJString()..deletedIn(arena);
      }, throwsRangeError);
      expect(() {
        array[3] = "44".toJString()..deletedIn(arena);
      }, throwsRangeError);
    });
  });
```

Reply:
I'll change it.